src/Fixer/Import/FullyQualifiedStrictTypesFixer.php

```diff
@@ -0,0 +1,226 @@
+            if ($shortType === $argument->getType()) {
+                continue;
+            }
+
+            $tokens->overrideRange(
+                $argument->getTypeIndexStart(),
+                $argument->getTypeIndexEnd(),
+                $this->generateTokensForShortType($shortType)
+            );
+        }
+    }
+
+    /**
+     * @param Tokens                $tokens
+     * @param int                   $index
+     * @param array<string, string> $namespaces a list of all FQN namespaces in the file with the short name as key
+     * @param array<string, string> $useMap     a list of all FQN use statements in the file with the short name as key
+     */
+    private function fixFunctionReturnType(Tokens $tokens, $index, array $namespaces, array $useMap)
+    {
+        if (PHP_VERSION_ID < 70000) {
+            return;
+        }
+
+        $returnType = (new FunctionsAnalyzer())->getFunctionReturnType($tokens, $index);
+        if (!$returnType) {
+            return;
+        }
+
+        $shortType = $this->detectShortType($returnType->getType(), $namespaces, $useMap);
+        if ($shortType === $returnType->getType()) {
+            return;
+        }
+
+        $tokens->overrideRange(
+            $returnType->getStartIndex(),
+            $returnType->getEndIndex(),
+            $this->generateTokensForShortType($shortType)
+        );
+    }
+
+    /**
+     * @param string                $type
+     * @param array<string, string> $namespaces a list of all FQN namespaces in the file with the short name as key
+     * @param array<string, string> $useMap     a list of all FQN use statements in the file with the short name as key
+     *
+     * @return string
+     */
+    private function detectShortType($type, array $namespaces, array $useMap)
+    {
+        // First match explicit imports:
+        foreach ($useMap as $shortName => $fullName) {
+            $regex = '/^\\\\?'.preg_quote($fullName, '/').'$/';
+            if (preg_match($regex, $type)) {
+                return $shortName;
+            }
+        }
+
+        // Next try to match (partial) classes inside the same namespace
+        // For now only support one namespace per file:
+        if (1 === count($namespaces)) {
+            foreach ($namespaces as $shortName => $fullName) {
+                $matches = [];
+                $regex = '/^\\\\?'.preg_quote($fullName, '/').'\\\\(?P<className>.+)$/';
+                if (preg_match($regex, $type, $matches)) {
+                    return $matches['className'];
+                }
+            }
+        }
+
+        // Next: Try to match partial use statements:
+        foreach ($useMap as $shortName => $fullName) {
+            $matches = [];
+            $regex = '/^\\\\?'.preg_quote($fullName, '/').'\\\\(?P<className>.+)$/';
+            if (preg_match($regex, $type, $matches)) {
+                return $shortName.'\\'.$matches['className'];
+            }
+        }
+
+        return $type;
+    }
```

Comment:
This might deservce own class. It would be useable out of this fixer for other classes and also for writing own fixers.
I'd definitelly use it :+1:

Reply:
Agree. This would also avoid passing the namespaces and uses across the fixer.
Not sure where to place it though. Any suggestions?

Reply:
Great! I'm not exactly sure, what Analyzers are for, but if it would cover this, why not there:

```php
PhpCsFixer\Tokenizer\Analyzer\ShortNameAnalyzer
```

or

```php
PhpCsFixer\Tokenizer\Resolver\Namespace\ShortNameResolver
```

Reply:
Great, I'll go for the resolver since it is more then an analysis